text/0000-nested-publish.md

```diff
@@ -0,0 +1,303 @@
+
+  This check is intended only to prevent accidents (such as vendoring a third-party package without considering the implications of redistributing it). It is always valid to omit `package.license` from the nested package, thus making no machine-readable claims about its licensing independent from the parent package's.
+
+* We might want to explicitly prohibit nested packages from specifying a `package.version`, to avoid giving the misleading impression that it means anything. (`package.version` is already optional as of Cargo 1.75, but this is currently equivalent to `version = "0.0.0"`.)
+
+## **`cargo package` &amp; `cargo publish`**
+
+If `cargo package` or `cargo package` is directed to operate on a package whose manifest specifies `package.publish = "nested"`, that is an error. (Users wishing to make crates in a nested package public should either write `package.publish = true` or should re-export their contents from another package.)
+
+When a valid parent package is packaged, each of its transitive nested dependencies must be included in the `.crate` archive file. This has two sub-cases:
+
+* The nested package may be in a subdirectory of the parent package directory. In this case, it is copied to the same location in the archive.
+* Otherwise, it is copied to `.cargo/packages/<package name>/` within the archive.
+
+`Cargo.toml` files for nested packages are rewritten in the same way as is already done for all packages, except that `path` dependencies which are nested dependencies are kept rather than stripped out or rejected as they currently are. Their path values may need to be rewritten to point to the nested paackages' new location in the archive.
+
+## **`crates.io`**
+
+`crates.io` will allow uploading of packages that contain `path` dependencies that were previously prohibited, as long as:
+
+* The dependency is a valid nested dependency as defined above. This includes that the the named package in fact exists in the `.crate` archive file, and has a valid `Cargo.toml` which declares `package.publish = "nested"`.
+* The `path`s in all contained manifests must not contain any upward traversal outside of the parent package (`../../`) or other hazardous or non-portable components as determined to be necessary.
+
+The package index, and the `crates.io` user interface, does not explicitly represent nested packages; instead, nested packages' dependencies are flattened into the dependencies of the parent package when they are added to the index. This adequately reflects what can be expected when using the parent package. (Note that required dependencies of optional nested packages should become optional in the flattened form.)
```

Comment:
we need to include flattening the effect of features

Reply:
Done.